src/ui/public/vis/editors/default/default.js

```diff
@@ -64,19 +64,24 @@ const defaultEditor = function ($rootScope, $compile) {
 
           updateScope();
 
+          $scope.state = $scope.vis.copyCurrentState();
+          $scope.oldState = $scope.vis.getSerializableState($scope.state);
+
           $scope.toggleSidebar = () => {
             $scope.$broadcast('render');
           };
 
           this.el.one('renderComplete', resolve);
-
           // track state of editable vis vs. "actual" vis
           $scope.stageEditableVis = () => {
+            $scope.vis.setCurrentState($scope.state);
+            $scope.oldState = $scope.vis.getSerializableState($scope.state);
             $scope.vis.updateState();
             $scope.vis.dirty = false;
           };
           $scope.resetEditableVis = () => {
-            $scope.vis.resetState();
+            $scope.state = $scope.vis.copyCurrentState();
+            $scope.oldState = $scope.vis.getSerializableState($scope.state);
```

Comment:
Do we need to set the `oldState` here? Shouldn't it be the state we are resetting to anyway?